utils_cv/detection/dataset.py

```diff
@@ -0,0 +1,220 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import os
+import math
+from pathlib import Path
+from random import randrange
+from typing import List, Tuple, Union
+
+import torch
+from torch.utils.data import Dataset, Subset
+import xml.etree.ElementTree as ET
+from PIL import Image
+
+from .plot import display_bounding_boxes, plot_grid
+from .model import get_transform
+
+
+class DetectionDataset(object):
```

Comment:
- If you copied (and slightly edited) this code from somewhere, can you provide a link. 
- We should brainstorm how this code here (copied and edited) can be best maintained. E.g. should we change the code to look nicer? Should we remove functions which we don't use, should we combine/refactor some of the functions (maybe not since that'll make it harder for us to copy newer updates of the code from the source in the future), etc.
- [minor and unsure] Should we add this functionality to data.py rather than having two .py files? classification does not have a dataset.py, maybe by putting everything into data.py we are more aligned?

Reply:
i did wrote this class, didn't copy it. It is loosely based on this: https://pytorch.org/tutorials/intermediate/torchvision_tutorial.html#writing-a-custom-dataset-for-pennfudan (which i will add as reference in a comment)

Reply:
also, all the functions in this class are used

Reply:
WRT to moving this functionality to data.py -- this is where i had initially written it in, but it got too big and unwieldy, which was why i moved it to a separate file.

Reply:
Thanks for the info!